test-suite/Math/NumberTheory/ArithmeticFunctions/InverseTests.hs

```diff
@@ -234,9 +320,60 @@ sigmaSpecialCase4 :: Assertion
 sigmaSpecialCase4 = assertBool "200 should be in inverseSigma(sigma(200))" $
   sigmaProperty1 $ Positive (200 :: Word)
 
+sigmas5 :: [Word]
+sigmas5 =
+  [ 1
+  , 33
+  , 244
+  , 1057
+  , 3126
+  , 8052
+  , 16808
+  , 33825
+  , 59293
+  , 103158
+  , 161052
+  , 257908
+  , 371294
+  , 554664
+  , 762744
+  , 1082401
+  , 1419858
+  , 1956669
+  , 2476100
+  , 3304182
+  , 4101152
+  , 5314716
+  , 6436344
+  , 8253300
+  , 9768751
+  , 12252702
+  , 14408200
+  , 17766056
+  , 20511150
+  ]
+
+sigmaSpecialCase5 :: [Assertion]
+sigmaSpecialCase5 = zipWith mkAssert ixs sigmas5
+ where
+  mkAssert a b = assertEqual "should be equal" (S.singleton a) (asSetOfPreimages (inverseSigmaK 5) b)
+  ixs = [1 .. 29]
+
 -------------------------------------------------------------------------------
 -- TestTree
 
+-- Tests for 'Int', 'Word' are omitted because 'inverseSigmaK/inverseJordan'
+-- tests would quickly oveflow in these types.
+testIntegralPropertyNoLargeInverse
+  :: forall wrapper1 wrapper2 bool. (TestableIntegral wrapper1, TestableIntegral wrapper2, SC.Testable IO bool, QC.Testable bool)
```

Comment:
It is probably worth to simplify this, inlining `wrapper1 = Power` and `wrapper2 = Positive`.